### PR TITLE
371158: app insight for adp-portal-api

### PR DESCRIPTION
**Pull request by rajeshkaremane**
# **What this PR does / why we need it**:
*A brief description of changes being made*
*Link to the relevant work items: *
[AB#371158](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/371158)

# **Special notes for your reviewer**
*Any specific actions or notes on review?*
Depends on https://github.com/DEFRA/adp-infrastructure-core/pull/173

# Testing
*Any relevant testing information and pipeline runs.*
https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=563698&view=results

tested in dev
image

tested locally
image

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [x] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: src/ADP.Portal.Api/AppInsightsTelemetryInitializer.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace ADP.Portal.Api
+{
+    public class AppInsightsTelemetryInitializer : ITelemetryInitializer
```
Comment (ppotturi) [inline src/ADP.Portal.Api/AppInsightsTelemetryInitializer.cs:6]: do we need this class?

Comment (rajeshkaremane) [reply src/ADP.Portal.Api/AppInsightsTelemetryInitializer.cs]: removed calls

File: .azuredevops/containerApp.yaml [hunk 2]
```diff
@@ -39,7 +39,10 @@ properties:
         identity: {{userAssignedIdentity}} 
       - name: git-repo-org
         keyVaultUrl: https://{{appKeyVaultName}}.vault.azure.net/secrets/GIT-REPO-ORG
-        identity: {{userAssignedIdentity}}         
+        identity: {{userAssignedIdentity}}
+      - name: app-insight-key
+        keyVaultUrl: https://{{appKeyVaultName}}.vault.azure.net/secrets/APP-INSIGHT-KEY
```
Comment (DaRoszko) [inline .azuredevops/containerApp.yaml:44]: Did we agree connection string orr Managed Identity for this?

Comment (rajeshkaremane) [reply .azuredevops/containerApp.yaml]: Code updated to use MI

File: src/ADP.Portal.Api/appsettings.Development.json [hunk 3]
```diff
@@ -5,6 +5,10 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
+  "AppInsights": {
+    "ConnectionString": "InstrumentationKey=b017c421-8e58-4f3e-8d3c-3f0e2974296a;IngestionEndpoint=https://uksouth-1.in.applicationinsights.azure.com/;LiveEndpoint=https://uksouth.livediagnostics.monitor.azure.com/;ApplicationId=208245fb-288b-45b9-b18f-07c2de588583",
+    "CloudRole": "adp-portal-api-local"
```
Comment (ppotturi) [inline src/ADP.Portal.Api/appsettings.Development.json:10]: I think we can remove this ?

Comment (rajeshkaremane) [reply src/ADP.Portal.Api/appsettings.Development.json:10]: updated

**Review by ppotturi — APPROVED** [review]

**Review by kbabigumira — APPROVED** [review]

**Pull request merged**